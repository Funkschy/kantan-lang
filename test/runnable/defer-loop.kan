import "io";

def f() {
    defer io.printf("end\n");
    let i = 0;
    while i < 10 {
        defer i += 1;
        defer io.printf("%d\n", i);
        if i % 2 == 0 {
            continue;
        }

        defer io.printf("uneven %d\n", i);
        if i > 4 {
            break;
        }
    }
}

def main() {
    f();
}
