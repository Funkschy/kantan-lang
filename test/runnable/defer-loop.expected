0
uneven 1
1
2
uneven 3
3
4
uneven 5
5
end
